File: src/main/java/com/github/ericufo/jedai/rag/RetrievedChunk.java
package com.github.ericufo.jedai.rag;

/**
 * 检索到的知识块，包含内容和来源信息
 */
public class RetrievedChunk {
    private final String content;
    private final String sourceDoc;
    private final Integer page;
    private final Integer[] pageRange;
    private final double score;

    public RetrievedChunk(String content, String sourceDoc, Integer page, Integer[] pageRange, double score) {
        this.content = content;
        this.sourceDoc = sourceDoc;
        this.page = page;
        this.pageRange = pageRange;
        this.score = score;
    }

    public RetrievedChunk(String content, String sourceDoc, Integer page, double score) {
        this(content, sourceDoc, page, null, score);
    }

    public String getContent() {
        return content;
    }

    public String getSourceDoc() {
        return sourceDoc;
    }

    public Integer getPage() {
        return page;
    }

    public Integer[] getPageRange() {
        return pageRange;
    }

    public double getScore() {
        return score;
    }
}

File: src/main/kotlin/com/github/ericufo/jedai/rag/RetrievedChunk.kt
package com.github.ericufo.jedai.rag

/**
 * 检索到的知识块，包含内容和来源信息
 */
data class RetrievedChunk(
    val content: String,
    val sourceDoc: String,        // 文件名或相对路径
    val page: Int?,               // 页码（如果支持）
    val pageRange: IntRange? = null,  // 页码范围（可选）
    val score: Double             // 相关性分数
)

